src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -629,6 +629,33 @@ private static Vector256<byte> ShuffleMaskShiftAlpha() => Vector256.Create((byte
             return Avx.Subtract(c, Avx.Multiply(a, b));
         }
 
+        /// <summary>
+        /// Blend packed 8-bit integers from <paramref name="left"/> and <paramref name="right"/> using <paramref name="mask"/>.
+        /// The high bit of each corresponding <paramref name="mask"/> byte determines the selection.
+        /// If the high bit is set the element of <paramref name="left"/> is selected.
+        /// The element of <paramref name="right"/> is selected otherwise.
+        /// </summary>
+        /// <param name="left">The left vector.</param>
+        /// <param name="right">The right vector.</param>
+        /// <param name="mask">The mask vector.</param>
+        /// <returns>The <see cref="Vector256{T}"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<byte> BlendVariable(in Vector128<byte> left, in Vector128<byte> right, in Vector128<byte> mask)
```

Comment:
```suggestion
        public static Vector128<byte> BlendVariable(Vector128<byte> left, Vector128<byte> right, Vector128<byte> mask)
```
`in` modifier is equivalent to `readonly ref`, so the vectors can't be passed in registers, rather they need to be passed via memory which is slower.
Due the _AggressiveInlining_ this method is likely to be inlined, so the JIT can figure out that no passing via memory is needed and can it can pass these vectors via registers -- but: that's not guaranteed, and the JIT has more (unnecessary) work to do.

I see this similar to a method like `int Add(int a, int b);` where one can assume that `a`, and `b` are passed by value and via registers at cpu-level. No need to put some emphasis on the read-only nature of `a`, and `b`. Some goes for vectors.

---

Side note: starting with .NET 7 this could just be `Vector128.ConditionalSelect` (I know we have a .NET 6 target, but I mention it for completness)

Reply:
Done